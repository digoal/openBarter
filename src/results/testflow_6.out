reset role;
SET client_min_messages = notice;
SET log_error_verbosity = terse;
-- SELECT ftruncatetables();
-- select fresetmarket();
SET role client;
-- two concurrent paths, (a) is the best
-- path (b)
-- select finsertorder('A','x',25  ,100 ,'z');
select finsertorder('A','z',200 ,200 ,'x');
NOTICE:  owner A created
CONTEXT:  PL/pgSQL function fgetowner(text,boolean) line 10 at assignment
PL/pgSQL function finsertorder(text,text,bigint,bigint,text) line 21 at assignment
          finsertorder          
--------------------------------
 (1,3-1,8,11,200,10,200,0,0,{})
(1 row)

-- path (a)
-- select finsertorder('B','x',25  ,200 ,'y');
select finsertorder('B','y',200 ,25  ,'x');
NOTICE:  owner B created
CONTEXT:  PL/pgSQL function fgetowner(text,boolean) line 10 at assignment
PL/pgSQL function finsertorder(text,text,bigint,bigint,text) line 21 at assignment
         finsertorder          
-------------------------------
 (2,3-2,9,11,25,12,200,0,0,{})
(1 row)

--select finsertorder('C','y',100 ,100 ,'z');
select finsertorder('C','z',100 ,100 ,'y');
NOTICE:  owner C created
CONTEXT:  PL/pgSQL function fgetowner(text,boolean) line 10 at assignment
PL/pgSQL function finsertorder(text,text,bigint,bigint,text) line 21 at assignment
          finsertorder           
---------------------------------
 (3,3-3,10,12,100,10,100,0,0,{})
(1 row)

-- no exchange
select id,uuid,nb,oruuid,grp,provider,quality,qtt,receiver from vmvt order by uuid;
 id | uuid | nb | oruuid | grp | provider | quality | qtt | receiver 
----+------+----+--------+-----+----------+---------+-----+----------
(0 rows)

--select finsertorder('D','z',250 ,250 ,'x');
select finsertorder('D','x',250 ,250 ,'z');
NOTICE:  owner D created
CONTEXT:  PL/pgSQL function fgetowner(text,boolean) line 10 at assignment
PL/pgSQL function finsertorder(text,text,bigint,bigint,text) line 21 at assignment
                                                                                               finsertorder                                                                                               
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 (4,3-4,11,10,250,11,250,300,250,"{""[(2, 9, 11, 25, 12, 200, 200),(3, 10, 12, 100, 10, 100, 100),(4, 11, 10, 250, 11, 250, 250)]"",""[(1, 8, 11, 200, 10, 200, 200),(4, 11, 10, 250, 11, 250, 200)]""}")
(1 row)

--two exchanges in a single transaction
select id,uuid,nb,oruuid,grp,provider,quality,qtt,receiver from vmvt order by uuid;
 id | uuid | nb | oruuid | grp | provider | quality | qtt | receiver 
----+------+----+--------+-----+----------+---------+-----+----------
  1 | 3-1  |  3 | 3-4    | 3-1 | D        | x       |  50 | B
  2 | 3-2  |  3 | 3-2    | 3-1 | B        | y       | 200 | C
  3 | 3-3  |  3 | 3-3    | 3-1 | C        | z       | 100 | D
  4 | 3-4  |  2 | 3-4    | 3-4 | D        | x       | 200 | A
  5 | 3-5  |  2 | 3-1    | 3-4 | A        | z       | 200 | D
(5 rows)

select * from fgetstats(true);
            _name            | cnt 
-----------------------------+-----
 number of qualities         |  12
 number of owners            |  11
 number of quotes            |   0
 number of orders            |  28
 number of movements         |  31
 number of quotes removed    |   5
 number of orders removed    |  28
 number of movements removed |  26
 number of agreements        |  12
 number of orders rejected   |   0
 agreements with 2 partners  |   5
 agreements with 3 partners  |   7
(12 rows)

select * from fgeterrs();
            _name             | cnt 
------------------------------+-----
 errors on quantities in mvts |   0
 errors on agreements in mvts |   0
(2 rows)

