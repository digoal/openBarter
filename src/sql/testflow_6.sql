reset role;

-- SELECT ftruncatetables();
select fresetmarket();

-- two concurrent paths, (a) is the best
-- path (b)
-- select finsertorder('A','x',25  ,100 ,'z');
select finsertorder('A','z',200 ,200 ,'x');
-- path (a)
-- select finsertorder('B','x',25  ,200 ,'y');
select finsertorder('B','y',200 ,25  ,'x');
--select finsertorder('C','y',100 ,100 ,'z');
select finsertorder('C','z',100 ,100 ,'y');

-- no exchange
select id,nb,oruuid,grp,provider,quality,qtt,receiver from vmvt;

--select finsertorder('D','z',250 ,250 ,'x');
select finsertorder('D','x',250 ,250 ,'z');
--two exchanges in a single transaction
select id,nb,oruuid,grp,provider,quality,qtt,receiver from vmvt;

select id,qtt from tquality;
select * from fgetstats(true);

