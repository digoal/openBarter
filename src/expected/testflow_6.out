reset role;
SELECT ftruncatetables();
NOTICE:  truncate cascades to table "torder"
CONTEXT:  SQL statement "TRUNCATE tquality CASCADE"
PL/pgSQL function "ftruncatetables" line 11 at SQL statement
NOTICE:  truncate cascades to table "tmvt"
CONTEXT:  SQL statement "TRUNCATE tquality CASCADE"
PL/pgSQL function "ftruncatetables" line 11 at SQL statement
NOTICE:  truncate cascades to table "tmvtremoved"
CONTEXT:  SQL statement "TRUNCATE tquality CASCADE"
PL/pgSQL function "ftruncatetables" line 11 at SQL statement
NOTICE:  truncate cascades to table "torder"
CONTEXT:  SQL statement "TRUNCATE towner CASCADE"
PL/pgSQL function "ftruncatetables" line 13 at SQL statement
NOTICE:  truncate cascades to table "tmvt"
CONTEXT:  SQL statement "TRUNCATE towner CASCADE"
PL/pgSQL function "ftruncatetables" line 13 at SQL statement
NOTICE:  truncate cascades to table "tmvtremoved"
CONTEXT:  SQL statement "TRUNCATE towner CASCADE"
PL/pgSQL function "ftruncatetables" line 13 at SQL statement
INFO:  The command: VACUUM FULL ANALYZE is recommended
INFO:  All market tables have been truncated
 ftruncatetables 
-----------------
 
(1 row)

-- two concurrent paths, (a) is the best
-- path (b)
-- select finsertorder('A','x',25  ,100 ,'z');
select finsertorder('A','z',200 ,200 ,'x');
INFO:  owner A created
CONTEXT:  PL/pgSQL function "finsertorder" line 35 at assignment
 finsertorder 
--------------
(0 rows)

-- path (a)
-- select finsertorder('B','x',25  ,200 ,'y');
select finsertorder('B','y',200 ,25  ,'x');
INFO:  owner B created
CONTEXT:  PL/pgSQL function "finsertorder" line 35 at assignment
 finsertorder 
--------------
(0 rows)

--select finsertorder('C','y',100 ,100 ,'z');
select finsertorder('C','z',100 ,100 ,'y');
INFO:  owner C created
CONTEXT:  PL/pgSQL function "finsertorder" line 35 at assignment
 finsertorder 
--------------
(0 rows)

-- no exchange
select id,nb,oruuid,grp,provider,quality,qtt,receiver from vmvt;
 id | nb | oruuid | grp | provider | quality | qtt | receiver 
----+----+--------+-----+----------+---------+-----+----------
(0 rows)

--select finsertorder('D','z',250 ,250 ,'x');
select finsertorder('D','x',250 ,250 ,'z');
INFO:  owner D created
CONTEXT:  PL/pgSQL function "finsertorder" line 35 at assignment
 finsertorder 
--------------
 (3-4,3,1)
 (3-4,2,4)
(2 rows)

--two exchanges in a single transaction
select id,nb,oruuid,grp,provider,quality,qtt,receiver from vmvt;
 id | nb | oruuid | grp | provider | quality | qtt | receiver 
----+----+--------+-----+----------+---------+-----+----------
  2 |  3 | 3-2    |   1 | B        | y       | 200 | C
  3 |  3 | 3-3    |   1 | C        | z       | 100 | D
  1 |  3 | 3-4    |   1 | D        | x       |  50 | B
  5 |  2 | 3-1    |   4 | A        | z       | 200 | D
  4 |  2 | 3-4    |   4 | D        | x       | 200 | A
(5 rows)

select id,qtt from tquality;
 id | qtt 
----+-----
  3 | 200
  2 | 250
  1 | 300
(3 rows)

select * from fgetstats(true);
            _name             | cnt 
------------------------------+-----
 number of qualities          |   3
 number of owners             |   4
 number of orders             |   4
 number of movements          |   5
 number of orders removed     |   4
 number of movements removed  |   0
 total number of agreements   |   2
 errors on quantities in mvts |   0
 errors on agreements in mvts |   0
 MAXCYCLE                     |   8
 VERSION                      |  31
 INSERT_OWN_UNKNOWN           |   1
 CHECK_QUALITY_OWNERSHIP      |   0
 MAXORDERFETCH                | 100
 MARKET_OPENED                |   1
 MARKET_SESSION               |   3
 agreements with 2 partners   |   1
 agreements with 3 partners   |   1
(18 rows)

